### PR TITLE
Fix regression issue field indexes messed up for queries with fragments

### PR DESCRIPTION
**Pull request by sav007**
Fix regression issue introduced after adding support fragments with directives. When query contains fragments and inline fragments at the same level, indexes used to reference GraphQL response fields (in `ResponseField[]` array) get out of order. This happens because we removed `ResponseField.forString` code generation for surrogate field `Fragments`. This fix reverts back changes that caused index to be out of order.

For info see original issue, but this is where it gets messed up:
https://github.com/apollographql/apollo-android/pull/1994/files#diff-4784749789d3edec94f885e8663e6246R81

Because this was missing:
https://github.com/apollographql/apollo-android/pull/1994/files#diff-4784749789d3edec94f885e8663e6246R46

Closes https://github.com/apollographql/apollo-android/issues/1993

### CODE REVIEW

**Comment by sav007** [comment]
cc @designatednerd

**Comment by designatednerd** [comment]
Thanks! @pcarrier How's this lookin' to you?

**Comment by pcarrier** [comment]
@designatednerd I have to accept my limitations and admit that I have yet to grasp the design of this compiler and associated runtime decisions, so I'll happily defer to @sav007. Thanks so much for the quick turnaround; how quickly can we hope to land into published artifacts?

**Comment by pcarrier** [comment]
@designatednerd I have to accept my limitations and admit that I have yet to grasp the design of this compiler and its generated Kotlin code to reason effectively about its correctness, but this absolutely prevents the bug as we've observed it. Thanks so much for the quick turnaround; how quickly do we expect to land such bugfixes into published artifacts?

**Comment by designatednerd** [comment]
I'll leave code review to @martinbonnin and @tasomaniac, but it would be helpful to ship this ASAP since it's a regression.

File: apollo-compiler/src/test/graphql/com/example/unique_type_name/HeroDetailQuery.kt [hunk 1]
```diff
@@ -62,6 +62,7 @@ class HeroDetailQuery : Query<HeroDetailQuery.Data, HeroDetailQuery.Data, Operat
 
     companion object {
       private val RESPONSE_FIELDS: Array<ResponseField> = arrayOf(
+          ResponseField.forString("__typename", "__typename", null, false, null),
```
Comment (tasomaniac) [inline apollo-compiler/src/test/graphql/com/example/unique_type_name/HeroDetailQuery.kt]: Does it make sense that the same line is next to each other here?

Comment (sav007) [reply apollo-compiler/src/test/graphql/com/example/unique_type_name/HeroDetailQuery.kt]: Do you mean the same line of  `ResponseField.forString("__typename", "__typename", null, false, null)`?  If yes that's correct, first line used by actual field `__typename` and the second one is not involved in parsing response, but it needs to be there to have right field index.

I'm thinking of making changes to remove `RESPONSE_FIELDS ` all together, as index based access to the response field meta looks fragile.

**Pull request merged**